src/commands/generateFunctions.ts

```diff
@@ -0,0 +1,104 @@
+import { OpenDialogOptions, ProgressLocation, Uri, window, workspace } from "vscode";
+import { ApiTreeItem } from "../explorer/ApiTreeItem";
+import { OpenApiEditor } from "../explorer/editors/openApi/OpenApiEditor";
+import { ext } from "../extensionVariables";
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { localize } from "../localize";
+import { cpUtils } from "../utils/cpUtils";
+
+// tslint:disable: no-non-null-assertion
+export async function generateFunctions(node?: ApiTreeItem): Promise<void> {
+    if (!node) {
+        node = <ApiTreeItem>await ext.tree.showTreeItemPicker(ApiTreeItem.contextValue);
+    }
+
+    ext.outputChannel.show();
+
+    const openApiEditor: OpenApiEditor = new OpenApiEditor();
+    const openAPIDocString = await openApiEditor.getData(node);
+
+    const languages = ['Python', 'CSharp', 'TypeScript', 'Java'];
+    const language = await ext.ui.showQuickPick(
+        languages.map((s) => {return { label: s, description: '', detail: '' }; }), { placeHolder: "Select language", canPickMany: false});
+
+    const uris = await askFolder();
+
+    const openAPIFilePath = path.join(uris[0].fsPath, `${node!.apiContract.name}.json`);
+
+    await window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("openAPI", `Downloading OpenAPI Document for API '${node.apiContract.name}'...`),
+            cancellable: false
+        },
+        async () => {
+            await fse.writeFile(openAPIFilePath, openAPIDocString);
+        }
+    ).then(async () => {
+        window.showInformationMessage(localize("openAPIDownloaded", `Downloaded OpenAPI Document for API '${node!.apiContract.name} successfully.`));
+    });
+
+    const autoRestInstalled = await isAutoRestInstalled();
+    if (!autoRestInstalled) {
+        await window.withProgress(
+            {
+                location: ProgressLocation.Notification,
+                title: localize("installAutorest", `Installing autorest...`),
```

Comment:
Adding some reference to what autorest is would help.

```suggestion
                title: localize("installAutorest", `Installing AutoRest (http://aka.ms/autorest)...`),
```